mathics/eval/testing_expressions.py

```diff
@@ -99,3 +101,64 @@ def expr_min(elements):
 
 def is_number(sympy_value) -> bool:
     return hasattr(sympy_value, "is_number") or isinstance(sympy_value, sympy.Float)
+
+
+def check_ArrayQ(expr, pattern, test, evaluation: Evaluation):
```

Comment:
Thanks for moving this out of mathics/builtin/testing_expressions/list_oriented.py

Reply:
@Li-Xiang-Ideal,  I think that just this step is valuable to be put alone in a PR.

Reply:
I agree that this is probably enough of a first independent step if the goal is to fix up SparseArray. It is a reasonable-size PR.